### ISSUE TITLE
where is config/environment.Lamby.cmd ?

### ISSUE THREAD

**Issue by hopewise** [rails-lambda/lamby-cookiecutter#25]
Hello

I have tried version 4 to deploy a new rails app using steps at https://lamby.custominktech.com/docs/quick-start, I got this error below right after deployment:

```
Executing 'config/environment.Lamby.cmd' in function directory '/app'
--
  | 2022-12-26T16:19:58.668+03:00CopyInit error when loading handler | Init error when loading handler
  | 2022-12-26T16:19:58.668+03:00 | {
  | 2022-12-26T16:19:58.668+03:00 | "errorMessage": "cannot load such file -- mail/indifferent_hash",
  | 2022-12-26T16:19:58.668+03:00 | "errorType": "Init<LoadError>",
  | 2022-12-26T16:19:58.668+03:00 | "stackTrace": [
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/kernel.rb:38:in `require'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/kernel.rb:38:in `require'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/mail-2.8.0/lib/mail.rb:14:in `<module:Mail>'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/mail-2.8.0/lib/mail.rb:3:in `<top (required)>'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/kernel.rb:38:in `require'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/kernel.rb:38:in `require'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/actionmailer-7.0.4/lib/action_mailer/base.rb:3:in `<top (required)>'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/kernel.rb:38:in `require'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/kernel.rb:38:in `require'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/app/mailers/application_mailer.rb:1:in `<top (required)>'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/kernel.rb:30:in `require'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/kernel.rb:30:in `require'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/loader/helpers.rb:135:in `const_get'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/loader/helpers.rb:135:in `cget'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/loader/eager_load.rb:169:in `block in actual_eager_load_dir'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/loader/helpers.rb:40:in `block in ls'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/loader/helpers.rb:25:in `each'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/loader/helpers.rb:25:in `ls'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/loader/eager_load.rb:164:in `actual_eager_load_dir'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/loader/eager_load.rb:17:in `block (2 levels) in eager_load'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/loader/eager_load.rb:16:in `each'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/loader/eager_load.rb:16:in `block in eager_load'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/loader/eager_load.rb:10:in `synchronize'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/loader/eager_load.rb:10:in `eager_load'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/loader.rb:296:in `block in eager_load_all'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/loader.rb:294:in `each'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.6/lib/zeitwerk/loader.rb:294:in `eager_load_all'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/railties-7.0.4/lib/rails/application/finisher.rb:74:in `block in <module:Finisher>'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/railties-7.0.4/lib/rails/initializable.rb:32:in `instance_exec'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/railties-7.0.4/lib/rails/initializable.rb:32:in `run'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/railties-7.0.4/lib/rails/initializable.rb:61:in `block in run_initializers'",
  | 2022-12-26T16:19:58.668+03:00 | "/usr/local/lib/ruby/3.1.0/tsort.rb:228:in `block in tsort_each'",
  | 2022-12-26T16:19:58.668+03:00 | "/usr/local/lib/ruby/3.1.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'",
  | 2022-12-26T16:19:58.668+03:00 | "/usr/local/lib/ruby/3.1.0/tsort.rb:431:in `each_strongly_connected_component_from'",
  | 2022-12-26T16:19:58.668+03:00 | "/usr/local/lib/ruby/3.1.0/tsort.rb:349:in `block in each_strongly_connected_component'",
  | 2022-12-26T16:19:58.668+03:00 | "/usr/local/lib/ruby/3.1.0/tsort.rb:347:in `each'",
  | 2022-12-26T16:19:58.668+03:00 | "/usr/local/lib/ruby/3.1.0/tsort.rb:347:in `call'",
  | 2022-12-26T16:19:58.668+03:00 | "/usr/local/lib/ruby/3.1.0/tsort.rb:347:in `each_strongly_connected_component'",
  | 2022-12-26T16:19:58.668+03:00 | "/usr/local/lib/ruby/3.1.0/tsort.rb:226:in `tsort_each'",
  | 2022-12-26T16:19:58.668+03:00 | "/usr/local/lib/ruby/3.1.0/tsort.rb:205:in `tsort_each'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/railties-7.0.4/lib/rails/initializable.rb:60:in `run_initializers'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/vendor/bundle/ruby/3.1.0/gems/railties-7.0.4/lib/rails/application.rb:372:in `initialize!'",
  | 2022-12-26T16:19:58.668+03:00 | "/app/config/environment.rb:5:in `<top (required)>'",
  | 2022-12-26T16:19:58.668+03:00 | "/usr/local/bundle/gems/aws_lambda_ric-2.0.0/lib/aws_lambda_ric.rb:41:in `require'",
  | 2022-12-26T16:19:58.668+03:00 | "/usr/local/bundle/gems/aws_lambda_ric-2.0.0/lib/aws_lambda_ric.rb:41:in `run'",
  | 2022-12-26T16:19:58.668+03:00 | "/usr/local/bundle/gems/aws_lambda_ric-2.0.0/lib/aws_lambda_ric/bootstrap.rb:35:in `bootstrap_handler'",
  | 2022-12-26T16:19:58.668+03:00 | "/usr/local/bundle/gems/aws_lambda_ric-2.0.0/lib/aws_lambda_ric/bootstrap.rb:8:in `start'",
  | 2022-12-26T16:19:58.668+03:00 | "/usr/local/bundle/gems/aws_lambda_ric-2.0.0/bin/aws_lambda_ric:10:in `<top (required)>'",
  | 2022-12-26T16:19:58.668+03:00 | "/usr/local/bundle/bin/aws_lambda_ric:25:in `load'",
  | 2022-12-26T16:19:58.668+03:00 | "/usr/local/bundle/bin/aws_lambda_ric:25:in `<main>'"
  | 2022-12-26T16:19:58.668+03:00 | ]

```

Where is `config/environment.Lamby.cmd` ? shouldn't it work out of the box as per the documentation ?

**Comment by metaskills** [comment]
UGH... the error is `"errorMessage": "cannot load such file -- mail/indifferent_hash"` and this is the reason you hit the error:

- https://github.com/mikel/mail/issues/1541

Please add this to your `Gemfile` to work around it.

```ruby
gem 'mail', '2.7.1'
```

Also, to answer your question. This `CMD` string `config/environment.Lamby.cmd` is broken up in two parts by the RIC (Runtime Interface Client). The first part is `config/environment` which is expanded by the RIC to `config/environment.rb` which is a file in your Rails app that loads the Rails application. The second is the method `Lamby.cmd` which is defined in the lamby gem here. https://github.com/customink/lamby/blob/master/lib/lamby.rb#L23-L27 Does that help?

Your core issue is the mail gem tho. So sorry you hit that. Been trying to get the Mail gem folks to fix their stuff for a few weeks now :(

**Comment by metaskills** [comment]
I just merged #26 to help others not hit this. Sorry you got caught up in that. Lemme know how things work out for you.

**Comment by hopewise** [comment]
Thanks @metaskills , that has fixed the issue 👍 By the way, do you have any recommendation for similar solution to Lamby for NodeJS ex: NuxtJS app ?
